corehq/apps/case_search/utils.py

```diff
@@ -400,12 +398,36 @@ def get_related_cases_result(helper, app, case_types, source_cases, include_all_
         return _get_all_related_cases(helper, source_cases)
     else:
         results = []
-        results.extend(_get_search_detail_path_defined_cases(helper, app, case_types, source_cases))
-        source_case_ids = {case.case_id for case in source_cases}
-        results.extend(_get_child_cases_referenced_in_app(helper, app, case_types, source_case_ids))
+        with helper.profiler.timing_context('get_app_context'):
+            paths, child_case_types = get_app_context(helper.domain, app_id, case_types)
+        if paths:
```

Comment:
Previously, this called `_get_search_detail_path_defined_cases`, which called `get_path_related_cases_results` only `if paths`.  When I removed the rest of `_get_search_detail_path_defined_cases`, I inlined the `if`.  Likewise for `_get_child_cases_referenced_in_app` and `get_child_case_results`.

It does look like `get_path_related_cases_results` would noop if I dropped the `if paths`, so that should be fine, but `get_child_case_results` doesn't.  Still, I think I prefer to leave the `if` in there even in the former case, as that's a profiled function, and this makes it a little clearer what portions of the code are executed.